lib/crypt.cpp

```diff
@@ -349,6 +349,7 @@ int check_file_signature(
     char clear_buf[MD5_LEN];
     int n, retval;
     DATA_BLOCK clear_signature;
+    clear_buf[0]=0;
```

Comment:
It is an output variable.
If OpenSSL reads smth from output variable - it'd weird.
I'm not sure this is necessary here, for me looks like false positive detection

Reply:
It is about the line that assigns a value to "answer" in https://github.com/BOINC/boinc/blob/fe1ddbafa7735160e5bd3587618aedbc527e957b/lib/crypt.cpp#L365
I know that a value should be assigned if retval is 0, but cppcheck does not know, so, yes, it is false positive, in a way, that is as long as you trust that function you called. I would nonetheless prefer to have it a bit more robust and see the patch in. Also, maybe you want to set the answer variable when bailing out early.